web_poet/rules.py

```diff
@@ -259,6 +317,34 @@ def search_overrides(self, **kwargs) -> List[ApplyRule]:
         warnings.warn(msg, DeprecationWarning, stacklevel=2)
         return self.search(**kwargs)
 
+    def _rules_for_url(
+        self, url: Union[_Url, str], matchers: Dict[Any, URLMatcher]
+    ) -> Mapping[Type, Type[ItemPage]]:
+        result: Dict[Type, Type[ItemPage]] = {}
+
+        url = str(url)
+
+        for target, matcher in matchers.items():
+            rule_id = matcher.match(url)
+            if rule_id is not None:
+                result[target] = self._rules[rule_id].use
+
+        return result
+
+    def overrides_for(self, url: Union[_Url, str]) -> Mapping[Type, Type[ItemPage]]:
```

Comment:
If that's a page object=>page object mapping, can we use `Mapping[Type[ItemPage], Type[ItemPage]]`?

Reply:
Done in https://github.com/scrapinghub/web-poet/pull/112/commits/c53249cfb8372dd520e26cb5b40f72dfa2aa0080